### DISCUSSION TITLE
Add Video Playback Tracks to AnimationPlayer

### DISCUSSION THREAD

**Discussion by skeddles** [godotengine/godot-proposals#10582] [Editor]
Right now you can play audio clips with an AnimationPlayer node by adding a track and selecting "Audio Playback Track...", then dragging in audio clips which you can just move or shorten.

It would be nice if you could do the same with video clips, so you could play certain clips from a video file at certain times

**Comment by AThousandShips** [comment]
Where would these be rendered? Audio makes sense to not require something to display on, but how would a video be drawn?

To accomplish this you can just add a custom method to your node where you want to play it, and have this method just take a video stream resource and play it when this method is called, and then add a method call track to the animation, or even add a property to the node and have the setter play the video, then you can just use a property track

**Reply by theraot** [reply]
When you create an audio play back track, you need to provide AudioStreamPlayer. This makes sense, for example, since Godot supports 3D audio, so it matters where it plays. So a video playback track would similarly require a VideoStreamPlayer.

To ochestrate the VideoStreamPlayer to play multiple videos, you need to set the stream and call play (the play method does not take the stream as a parameter). Yes, this can be done from the AnimationPlayer, or we can write a method to do it and call the method from the AnimationPlayer... And then we need to be careful to do that after the prior video ended.

An audio playback track solves that for AudioStreams (it plays the requested stream, and allows you to see the duration of the audio on the animation editor). I understand that the usecase for VideoStreams is much less common. Yet, I believe this proposal makes sense.

**Reply by AThousandShips** [reply]
That's true I forgot that and misremembered it as the player handled the audio playback rather than that it just handled the polyphony itself
